File: Eleven.java
import java.util.Scanner;

public class Eleven {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] heights = new int[n];
        for (int i = 0; i < n; i++) {
            heights[i] = scanner.nextInt();
        }
        System.out.println(maxWater(heights, n));
    }

    public static int maxWater(int[] heightsArr, int n) {
        int[] leftMax = new int[n];
        int[] rightMax = new int[n];

        leftMax[0] = heightsArr[0];
        for (int i = 1; i < n; i++) {
            leftMax[i] = Math.max(leftMax[i - 1], heightsArr[i]);
        }

        rightMax[n - 1] = heightsArr[n - 1];
        for (int i = n - 2; i >= 0; i--) {
            rightMax[i] = Math.max(rightMax[i + 1], heightsArr[i]);
        }

        int waters = 0;
        for (int i = 0; i < n; i++) {
            int minHeight = Math.min(leftMax[i], rightMax[i]);
            if (minHeight > heightsArr[i]) {
                waters += minHeight - heightsArr[i];
            }
        }

        return waters;
    }
}

File: eleven.py

def max_water(heights_arr, n):

    left_max = n * [-1]
    right_max = n * [-1]

    left_max[0] = heights_arr[0]
    for i in range(1, n):
        left_max[i] = max(left_max[i-1], heights_arr[i])

    right_max[n-1] = heights_arr[n-1]
    for i in range(n-2, -1, -1):
        right_max[i] = max(right_max[i+1], heights_arr[i])

    waters = 0
    for i in range(n):
        min_height = min(left_max[i], right_max[i])
        if min_height > heights_arr[i]:
            waters += min_height - heights_arr[i]
    return waters


n = int(input())
heights = list(map(int, input().split()))
print(max_water(heights , n))
